### ISSUE TITLE
`iris.cube.Cube.regrid()`-compatible scheme for `MeshToGridESMFRegridder`

### ISSUE THREAD

**Issue by schlunma** [SciTools-incubator/iris-esmf-regrid#192]
## ✨ Feature Request

It would be great to have a regridding scheme that can be recognized by [`iris.cube.Cube.regrid()`](https://scitools-iris.readthedocs.io/en/latest/generated/api/iris/cube.html#iris.cube.Cube.regrid) based on [`MeshToGridESMFRegridder`](https://iris-esmf-regrid.readthedocs.io/en/latest/_api_generated/esmf_regrid.experimental.unstructured_scheme.html#esmf_regrid.experimental.unstructured_scheme.MeshToGridESMFRegridder) (just like [`ESMFAreaWeighted`](https://iris-esmf-regrid.readthedocs.io/en/latest/_api_generated/esmf_regrid.schemes.html#esmf_regrid.schemes.ESMFAreaWeighted) for [`ESMFAreaWeightedRegridder`](https://iris-esmf-regrid.readthedocs.io/en/latest/_api_generated/esmf_regrid.schemes.html#esmf_regrid.schemes.ESMFAreaWeightedRegridder)).

## Motivation

In ESMValTool, we use [`iris.cube.Cube.regrid()` for regridding](https://github.com/ESMValGroup/ESMValCore/blob/551bd3092110cadfe8159544747e969625ecae1a/esmvalcore/preprocessor/_regrid.py#L636); thus, a mesh-to-grid (or the other way round) scheme that can be recognized by this function would be great!

Thanks for all the great work on this package!

**Comment by stephenworsley** [comment]
I'm looking to get this done. I'm thinking that ultimately, the unstructured and structured regridders could end up getting wrapped by the same scheme, `ESMFAreaWeighted` (and also `ESMFLinear`). To that end, I've been unifying the behaviour of the structured and unstructured regridders which had drifted due to being developed as part of separate projects. I've made some progress with #175 so hopefully I should be able to add this soon.

**Comment by schlunma** [comment]
That sounds great @stephenworsley!!

**Comment by schlunma** [comment]
still relevant

**Comment by schlunma** [comment]
still relevant

**Comment by stephenworsley** [comment]
@schlunma you should be aware that with #198, it is not possible to do mesh-to-grid regridding with the schemes `ESMFAreaWeighted` and `ESMFBilinear`, I'm hoping to do a release soon to get this available.

**Comment by schlunma** [comment]
@stephenworsley Thanks, this is awesome! Does that mean that `ESMFAreaWeighted` and `ESMFBilinear` will work for both mesh-to-grid and grid-to-grid regridding?

**Comment by stephenworsley** [comment]
@schlunma Yes, they'll work for mesh-to-grid, grid-to-grid and gid-to-mesh regridding, though we have not yet implemented mesh-to-mesh regridding.

**Comment by schlunma** [comment]
Fantastic, this is all we need! I'll give it a try as soon as your new release is out :rocket:
